@xen-orchestra/fs/src/s3.js

```diff
@@ -232,14 +230,17 @@ export default class S3Handler extends RemoteHandlerAbstract {
   }
 
   async _createReadStream(path, options) {
-    if (!(await this._isFile(path))) {
-      const error = new Error(`ENOENT: no such file '${path}'`)
-      error.code = 'ENOENT'
-      error.path = path
-      throw error
+    try {
+      return (await this._s3.send(new GetObjectCommand(this._createParams(path)))).Body
+    } catch (e) {
+      if (e.name === 'NoSuchKey') {
+        const error = new Error(`${path} does not exists`)
+        error.code = 'ENOENT'
+        error.path = path
+        throw error
+      }
+      throw e
```

Comment:
You can rebased on top of `master` and you should no longer have changes in this PR.